server/docs/getting-started.md

```diff
@@ -142,384 +288,97 @@ async function initWeb3() {
+----------
 
-```solidity
-function highRoller(bytes32 randomness)
-    public
-    pure
-    returns(uint8 playerFirstTotal, uint8 playerSecondTotal)
-  {
-    (bytes8 hash1, bytes8 hash2,
-    bytes8 hash3, bytes8 hash4) = cutBytes32(randomness);
-    playerFirstTotal = bytes8toDiceRoll(hash1) + bytes8toDiceRoll(hash2);
-    playerSecondTotal = bytes8toDiceRoll(hash3) + bytes8toDiceRoll(hash4);
-  }
-```
+## setupCF()
 
-Back to the javascript.
+Here's what you'll find in `setupCF()`, and there's no need to change any of it. 
 
+```typescript
+async function setupCF() {
+  nodeProvider = new cf.NodeProvider();
+  await nodeProvider.connect();
+}
+```
 
 ----------
 
 
-# HighRoller.js
+## The install() Function
 
+This is where we begin developing **HighRoller**.
 
-----------
+### Counterfactual's Provider and AppFactory Objects
 
+The `install()` function will
+* reset game state
+* instantiate a Counterfactual `Provider`
+* instantiate a Counterfactual `AppFactory` instance.
 
-## encoding, initial state, and resetGameState()
+The `Provider` is our tool for listening and responding to state changes in a virtual channel: it only needs to be given the `nodeProvider`.
 
-Now that we know what HighRoller.sol looks like, we can fill in the encoding for the game:
+The `AppFactory` is how we create new virtual channels, and it needs to know:
+  * the address of the contract which provides the application logic (**HighRoller.sol**)
+  * the encodings for the application state and actions (to maintain the privacy of state and actions)
+  * which `Provider` can listen and propose updates to state.
 
+We'll input the contract address for **HighRoller.sol**, and fill in the details of the `Action` and `AppState` data types for the encodings.
 
 ```typescript
 async function install() {
   resetGameState();
+  
+  const contractAddress = '0x91907355C59BA005843E791c88aAB80b779446c9';
+  const actionEncoding = 'tuple(uint8 actionType, uint256 number, bytes32 actionHash)';
+  const stateEncoding = 'tuple(address[2] playerAddrs, uint8 stage, bytes32 salt, bytes32 commitHash, uint256 playerFirstNumber, uint256 playerSecondNumber)'
 
   let cfProvider = new cf.Provider(nodeProvider);
   let appFactory = new cf.AppFactory(contractAddress, {
-    actionEncoding: "tuple(uint8 actionType, uint256 number, bytes32 actionHash)",
-    stateEncoding: "tuple(address[2] playerAddrs, uint8 stage, bytes32 salt, bytes32 commitHash, uint256 playerFirstNumber, uint256 playerSecondNumber)"
+    actionEncoding,
+    stateEncoding
   }, cfProvider);
-
-  proposeInstall(appFactory);
-
-  cfProvider.on('installVirtual', onInstallEvent);
-  cfProvider.on('updateState', onUpdateEvent);
 }
 ```
 
-We can also describe the initialState, which consists of the two addresses and the Stage set to PRE_GAME:
+We’ll come back to define the function `resetGameState()` once we have a better sense of what that will entail.
```

Comment:
This is a very roundabout way of writing a guide. We introduce a file, say that we can "reset game state", reference a function called `resetGameState()` and then say we will define it when we know what it is.

Why confuse a reader this way?